### DISCUSSION TITLE
Checking whether user is part of a group (follow-up)

### DISCUSSION THREAD

**Discussion by shark2k** [DirectoryTree/LdapRecord#719] [Q&A]
Hi @stevebauman (or whoever can answer).

This is a follow-up to https://github.com/DirectoryTree/LdapRecord/discussions/490 (which did help me).

In that answer @stevebauman gave this as one possible solution:

```php
use LdapRecord\Models\ActiveDirectory\User;

$user = User::find('cn=John Doe,ou=Users,dc=local,dc=com');

if ($user->groups()->whereContains('cn', 'Accounting')->exists()) {
    // The user is apart of the "Accounting" group.
} else {
    // The user is *not* apart of the "Accounting" group.
}
```

I did not implement that solution and ended up using a different one (that is not really all that efficient but works).
Anyway, I recently tried to implement the above using the following code as a test:

```php
$employeeInfo = User::on('connection2')->where('samaccountname', '=', 'mine')->get()->first();
if ($employeeInfo->groups()->where('cn', 'contains', 'dog')->exists()) {
      echo "Contains dog";
} else {
      echo 'Does not contain dog';
}
```

I specifically used "dog" to check because I am not a member of any group that contains the word "dog" in the CN, so I expect this code to fail.
However, when I run the code it displays "Contains dog" which is not correct.

If I were to loop through all the groups I am a member of and do a stripos($groupName, 'dog') in an if/else statement, that would give me the result I expected.  This is currently what I do, but again it is not efficient and if I can get the above code to work, it should speed up execution of my code by a decent amount.

Thanks for any direction anyone can give me on what I am doing wrong/missing.

-Shark2k

**Comment by stevebauman** [comment]
Hi @shark2k,

What happens if you run:

```php
$groups = $employeeInfo->groups()->where('cn', 'contains', 'dog')->get();

dd($groups);
```

What are the results?

**Reply by shark2k** [reply]
Hi @stevebauman,

First, thanks for the quick reply!

I am not using Laravel, just straight PHP, so I did a var_dump($groups) instead.
So that said, doing the (plain PHP version) of what you asked, the var_dump is showing "object(LdapRecord\Models\Collection)#31 (2)" and then within that "object(LdapRecord\Models\ActiveDirectory\Group)#33 (19)" and I see the "exists" property(?) which has bool(true).

I can give paste the result if you need but only if it will help since I would want to obfuscate some of the info coming back if I post it publicly.

As I was typing this and looking at the result, I noticed that in the var_dump for $groups there was a property (sorry, I might be using the wrong term here again) for "connection" which has "connection2" as the value.  Out of curiosity, I decided to change my initial call for the $employeeInfo to use "connection1", which is limited to the scope of the subsidiary I work in (connection2 is one level up to include all the subsidiaries OU's in our AD environment since we can be in groups at our HQ level which do not get retrieved when using connection1, which you helped me resolve in another question I opened).

So, using connection1, when I try the code in my OP with the if/else statement, the result comes back with "Does not contain dog" as expected.
Looking at the var_dump when using connection2, I did a find for the word "dog" and nothing comes up, so I am curious as to why a result is coming back.

Please let me know what other information you require to be able to understand/figure out what is going on.
I am curious if I am doing something wrong, something is not working as expected or it is working as expected.

Thanks again!
-Shark2k

**Reply by stevebauman** [reply]
Do the resulting models contain "dog" anywhere in their CN attribute?

**Reply by shark2k** [reply]
I searched for the word "dog" on the page and there were no results.

If you are OK with it, I can email you the result from your request tomorrow.
I can put the title of this question as the subject.
I would prefer not to put the results publicly and also feel that obfuscating it would make it pointless to post since you would not get to see the whole thing.

Let me know and/or if you have anything else you want me to try.

Thanks,
-Shark2k

**Reply by stevebauman** [reply]
Ok thanks for checking. Can you dump the query being executed and post it here?

```php
$query = $employeeInfo->groups()->where('cn', 'contains', 'dog')->getQuery()->getUnescapedQuery();

var_dump($query);
```

**Reply by shark2k** [reply]
Sure thing.
Here is the result:

```
string(10) "(cn=*dog*)"
```

Let me know if there is anything else I can provide to help with troubleshooting.

Thanks,
-Shark2k

**Reply by shark2k** [reply]
Hi @stevebauman,

I am not sure if you are trying to look into this to figure out what is going on or if you are just not sure and do not have anything else to add (which I am OK with and completely understand).

After my last reply I started looking through the documentation again and saw/remembered the setConnection method.
I ended up doing a test where I made the initial call for the users using connection2 and then I switched to connection1 and checked to see if the 'cn' contains the value.  As I mentioned in my first reply to you, using connection1 to do the check worked, so this method of changing the connection to connection1 and then checking also worked.

I then did further validation doing a test with the actual initial call to get all the users I want and then within the loop to go through the user collection, I am setting the connection to connection1 and then checking for the existence of various groups with if/else statements.  Doing it this way, versus my original way of looping through the groups the member is in and do an if/else check on each group name, ended up going down to taking about 2 minutes to process vs the original method taking about 10 minutes (so definitely more efficient).

Anyway, thanks for your help and if you end up figuring out why it was not working with the larger scoped connection I would be interested to know.
Also, if you think of anything I can do to help with troubleshooting if you want to continue, just let me know.

Thanks,
-Shark2k
